okhttp/src/main/java/okhttp3/internal/cache/DiskLruCache.kt

```diff
@@ -13,371 +13,350 @@
+    synchronized(this@DiskLruCache) {
+      if (!initialized || closed) {
+        return@Runnable // Nothing to do
+      }
 
-        try {
-          if (journalRebuildRequired()) {
-            rebuildJournal();
-            redundantOpCount = 0;
-          }
-        } catch (IOException e) {
-          mostRecentRebuildFailed = true;
-          journalWriter = Okio.buffer(Okio.blackhole());
+      try {
+        trimToSize()
+      } catch (_: IOException) {
+        mostRecentTrimFailed = true
+      }
+
+      try {
+        if (journalRebuildRequired()) {
+          rebuildJournal()
+          redundantOpCount = 0
         }
+      } catch (_: IOException) {
+        mostRecentRebuildFailed = true
+        journalWriter = blackholeSink().buffer()
       }
     }
-  };
-
-  DiskLruCache(FileSystem fileSystem, File directory, int appVersion, int valueCount, long maxSize,
-      Executor executor) {
-    this.fileSystem = fileSystem;
-    this.directory = directory;
-    this.appVersion = appVersion;
-    this.journalFile = new File(directory, JOURNAL_FILE);
-    this.journalFileTmp = new File(directory, JOURNAL_FILE_TEMP);
-    this.journalFileBackup = new File(directory, JOURNAL_FILE_BACKUP);
-    this.valueCount = valueCount;
-    this.maxSize = maxSize;
-    this.executor = executor;
   }
 
-  public synchronized void initialize() throws IOException {
-    assert Thread.holdsLock(this);
+  init {
+    this.journalFile = File(directory, JOURNAL_FILE)
+    this.journalFileTmp = File(directory, JOURNAL_FILE_TEMP)
+    this.journalFileBackup = File(directory, JOURNAL_FILE_BACKUP)
+  }
+
+  @Synchronized @Throws(IOException::class)
+  fun initialize() {
+    assert(Thread.holdsLock(this))
 
     if (initialized) {
-      return; // Already initialized.
+      return // Already initialized.
     }
 
     // If a bkp file exists, use it instead.
     if (fileSystem.exists(journalFileBackup)) {
       // If journal file also exists just delete backup file.
       if (fileSystem.exists(journalFile)) {
-        fileSystem.delete(journalFileBackup);
+        fileSystem.delete(journalFileBackup)
       } else {
-        fileSystem.rename(journalFileBackup, journalFile);
+        fileSystem.rename(journalFileBackup, journalFile)
       }
     }
 
     // Prefer to pick up where we left off.
     if (fileSystem.exists(journalFile)) {
       try {
-        readJournal();
-        processJournal();
-        initialized = true;
-        return;
-      } catch (IOException journalIsCorrupt) {
-        Platform.get().log(WARN, "DiskLruCache " + directory + " is corrupt: "
-            + journalIsCorrupt.getMessage() + ", removing", journalIsCorrupt);
+        readJournal()
+        processJournal()
+        initialized = true
+        return
+      } catch (journalIsCorrupt: IOException) {
+        Platform.get().log(WARN, "DiskLruCache $directory is corrupt: ${journalIsCorrupt.message}" +
+            ", removing", journalIsCorrupt)
```

Comment:
might be worth putting log statement on the next line to have it all together

Reply:
Good call, fixed